### PR TITLE
[BE] SessionResources Endpoints

### LINKED ISSUE #15: CRUD endpoints for SessionResource
POST /session-resource: takes in the resource id and the session id in the request body. adds an entry in the bridge table
DELETE /session-resource: takes in the resource id and the session id in the request body. deletes the entry in the bridge table
GET /sessions/{session_id}/resource: get all of the resource (full objects, not just the ids) for a given session id. Include the full resource object along with the bridge table information.

As a reminder, please write tests for your code and update the openapi.yaml
Reference [#12 ](https://github.com/GenerateNU/specialstandard/issues/12) or the pinned image for the schema

### PR DESCRIPTION
**Pull request by aahiltn**
# Description

[Link to Ticket](https://github.com/GenerateNU/specialstandard/issues/15)

Please include a summary of the changes and the related issue. Please also
include relevant motivation, context, and images!

# How Has This Been Tested?

Please describe the tests that you manually ran to verify your changes (beyond any unit/integration tests written and ran).

### Postman (Basic Screenshots here)
image
image
image

Edge cases are tested with unit/integration/db tests. 

# Checklist

- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have written unit tests for my code and tested it manually
- [x] New and existing unit tests pass locally with my changes
- [x] I have updated the OpenAPI spec, if needed

closes #15

### CODE REVIEW

File: backend/internal/service/handler/session_resource/delete_session_resource.go [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+	if err != nil {
+		slog.Error("Failed to delete session_resource", "err", err)
+		errStr := err.Error()
+		switch {
+		case strings.Contains(errStr, "23503") || strings.Contains(errStr, "relationship not found"): // foreign key violation
+			return errs.NotFound("session or resource not found")
+		case strings.Contains(errStr, "connection refused"):
+			return errs.InternalServerError("Database Connection Error")
+		default:
+			return errs.InternalServerError("Failed to Create Session Resource")
```
Comment (michelleli04) [inline backend/internal/service/handler/session_resource/delete_session_resource.go:34]: Should be failed to delete...

**Review by michelleli04 — APPROVED** [review]
